install/kubernetes/cilium/templates/_hubble-generate-certs-job-spec.tpl

```diff
@@ -0,0 +1,77 @@
+          imagePullPolicy: {{ .Values.hubble.tls.auto.cronJob.image.pullPolicy }}
+          command:
+            - "/usr/bin/cilium-certgen"
+          {{/* Because this is executed as a job, we pass the values as command line args instead of via config map,
+                this allows users to inspect the values used in past runs by inspecting the completed pod */ -}}
+          args:
+            {{- if .Values.debug.enabled }}
+            - "--debug"
+            {{- end }}
+            {{- $hubbleCAProvided := and .Values.hubble.tls.ca.cert .Values.hubble.tls.ca.key -}}
+            {{- if $hubbleCAProvided }}
+            - "--hubble-ca-generate=false"
+            - "--hubble-ca-key-file=/var/lib/cilium/tls/hubble-ca/tls.key"
+            - "--hubble-ca-cert-file=/var/lib/cilium/tls/hubble-ca/tls.crt"
+            {{- else }}
+            - "--hubble-ca-generate=true"
+            - "--hubble-ca-validity-duration={{ $certValiditySecondsStr }}"
+            - "--hubble-ca-config-map-name=hubble-ca-cert"
+            - "--hubble-ca-config-map-namespace={{ .Release.Namespace }}"
+            {{- end }}
+            {{- if and .Values.hubble.tls.server.cert .Values.hubble.tls.server.key $hubbleCAProvided }}
+            - "--hubble-server-cert-generate=false"
+            {{- else }}
+            - "--hubble-server-cert-generate=true"
+            - "--hubble-server-cert-validity-duration={{ $certValiditySecondsStr }}"
+            - "--hubble-server-cert-secret-name=hubble-server-certs"
+            - "--hubble-server-cert-secret-namespace={{ .Release.Namespace }}"
+            {{- end }}
+            {{- if and .Values.hubble.relay.tls.client.cert .Values.hubble.relay.tls.client.key $hubbleCAProvided }}
+            - "--hubble-relay-client-cert-generate=false"
+            {{- else }}
+            - "--hubble-relay-client-cert-generate=true"
+            - "--hubble-relay-client-cert-validity-duration={{ $certValiditySecondsStr }}"
+            - "--hubble-relay-client-cert-secret-name=hubble-relay-client-certs"
+            - "--hubble-relay-client-cert-secret-namespace={{ .Release.Namespace }}"
+            {{- end }}
+            {{- if or (and .Values.hubble.relay.tls.server.cert .Values.hubble.relay.tls.server.key) (not .Values.hubble.relay.tls.server.enabled) }}
+            - "--hubble-relay-server-cert-generate=false"
+            {{- else if .Values.hubble.relay.tls.server.enabled }}
+            - "--hubble-relay-server-cert-generate=true"
+            - "--hubble-relay-server-cert-validity-duration={{ $certValiditySecondsStr }}"
+            - "--hubble-relay-server-cert-secret-name=hubble-relay-server-certs"
+            - "--hubble-relay-server-cert-secret-namespace={{ .Release.Namespace }}"
+            {{- end }}
+          volumeMounts:
+          {{- if $hubbleCAProvided }}
+            - mountPath: /var/lib/cilium/tls/hubble-ca
+              name: hubble-ca-secret
```

Comment:
nit: For consistency with other TLS secrets (ie: `hubble-server-certs`, `hubble-relay-client-certs`, `hubble-relay-server-certs`), I think the Hubble CA secret should be named `hubble-ca-certs` rather than `hubble-ca-secret`.

Reply:
I intentionally named did not choose `hubble-ca-certs` to avoid confusion with 
https://github.com/cilium/cilium/blob/e9cb43c03179af6f7cd5d0abb9974daeb5a21e5a/install/kubernetes/cilium/templates/hubble-ca-configmap.yaml#L5 :(

Reply:
I see. But `hubble-ca-cert` is a configmap, not a secret. Thus, I think it makes even more sense to name it `hubble-ca-certs` because the cert value in the configmap and the secret should actually be the same.

Reply:
I think I disagree. I personally find it more confusing and potentially error prone if they would share the same name. 

But in the end, it's personal preference is moot point, consistency with the rest of the existing code base should take priority over personal preferences.

I have not yet checked if there is prior art for this pattern in the rest of of Cilium, but I'm marking this as resolved for this PR: If we decide to use the same name, then we will have to rename the above config map as well, and that is a change unrelated to this PR. So if we decide to rename, then let's do that in a follow-up.